File: Flood Fill.cpp
class Solution {
public:
    vector<vector<int>> floodFill(vector<vector<int>>& image, int sr, int sc, int newColor) {
        vector<vector<int>> directions={{0,1},{1,0},{0,-1},{-1,0}};
        queue<pair<int,int>> q;
        bitset<51>visited[51];
        pair<int,int>p,k;
        p.first=sr;
        p.second=sc;
        q.push(p);
        int val=image[sr][sc];
        while(!q.empty())
        {
            p=q.front();
            q.pop();
            
            image[p.first][p.second]=newColor;
            visited[p.first][p.second]=1;
            for(auto x : directions)
            {
                if(p.first+x[0]>=0 && p.first+x[0]<image.size() && p.second+x[1]>=0 && p.second+x[1]<image[0].size() ){
                   if(image[p.first+x[0]][p.second+x[1]]==val && visited[p.first+x[0]][p.second+x[1]]==0)
                    {
                        k.first=p.first+x[0];
                        k.second=p.second+x[1];
                        q.push(k);
                    } 
                }
            } 
        }
        return image;
    }
};

File: Golang Solutions/Flood Fill.go
func floodFill(image [][]int, sr int, sc int, newColor int) [][]int {
    var directions = [4][2]int { {0,1},{1,0},{0,-1},{-1,0} }
    q := [][]int{}
    var visited[51][51] bool
    q=append(q,[]int{sr,sc})
    val:=image[sr][sc]
    for ;len(q)!=0;{
        image[q[0][0]][q[0][1]]=newColor
        visited[q[0][0]][q[0][1]]=true
        for _,x := range directions{
            if(q[0][0]+x[0]>=0 && q[0][0]+x[0]<len(image) && q[0][1]+x[1]>=0 && q[0][1]+x[1]<len(image[0]) ){
                   if(image[q[0][0]+x[0]][q[0][1]+x[1]]==val && visited[q[0][0]+x[0]][q[0][1]+x[1]]==false){
                       q=append(q,[]int{q[0][0]+x[0],q[0][1]+x[1]})
                    } 
            }
        } 
        q=q[1:]
    }
    return image
    
}